lambdas/accesstoken/src/main/java/uk/gov/di/ipv/cri/passport/accesstoken/validation/TokenRequestValidator.java

```diff
@@ -0,0 +1,117 @@
+                throw new ClientAuthenticationException(
+                        "Unknown client, no client_id value or client_assertion jwt found in request");
+            } else {
+                clientId = queryParams.get(CLIENT_ID_PARAM);
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(JWT)) {
+                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
+                    throw new ClientAuthenticationException(
+                            String.format(
+                                    "Missing client_assertion jwt for configured client '%s'",
+                                    clientId));
+                }
+            }
+        } else {
+            PrivateKeyJWT clientJwt;
+            try {
+                clientJwt = PrivateKeyJWT.parse(requestBody);
+
+                clientId = clientJwt.getClientID().getValue();
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(NONE)) {
+                    return;
+                }
+
+                verifier.verify(clientJwt, null, null);
+                validateExpiration(clientJwt.getJWTAuthenticationClaimsSet());
+            } catch (ParseException | InvalidClientException | JOSEException e) {
+                LOGGER.error("Validation of client_assertion jwt failed");
+                throw new ClientAuthenticationException(e);
+            }
+        }
+    }
+
+    private String getClientAuthenticationMethod(String clientId)
+            throws ClientAuthenticationException {
+        String clientAuthenticationMethod =
+                configurationService.getClientAuthenticationMethod(clientId);
+        if (clientAuthenticationMethod == null) {
+            LOGGER.error("Failed to find config for client ID: {}", clientId);
+            throw new ClientAuthenticationException(
+                    String.format("Config for client ID '%s' not found", clientId));
+        }
+        return clientAuthenticationMethod;
+    }
+
+    private void validateExpiration(JWTAuthenticationClaimsSet claimsSet)
+            throws InvalidClientException {
+        Date expirationTime = claimsSet.getExpirationTime();
+        String maxAllowedTtl = configurationService.getMaxClientAuthTokenTtl();
+
+        OffsetDateTime offsetDateTime =
+                OffsetDateTime.now().plusSeconds(Long.parseLong(maxAllowedTtl));
+        if (expirationTime.getTime() / 1000L > offsetDateTime.toEpochSecond()) {
+            LOGGER.error("Client JWT expiry date is too far in the future");
+            throw new InvalidClientException(
+                    "The client JWT expiry date has surpassed the maximum allowed ttl value");
+        }
+    }
+
+    public static ClientAuthenticationVerifier<Object> getClientAuthVerifier(
```

Comment:
I think this doesn't need to be static as it's being called in the constructor directly.